packages/drafts-realm/event.gts

```diff
@@ -0,0 +1,251 @@
+        height: var(--boxel-form-control-height);
+        border-radius: var(--boxel-form-control-border-radius);
+        transition: border-color var(--boxel-transition);
+        border: 1px solid var(--boxel-form-control-border-color);
+        padding: var(--boxel-sp-xs) 0 var(--boxel-sp-xs) var(--boxel-sp-sm);
+        background-color: white;
+      }
+    </style>
+  </template>
+}
+
+class Isolated extends Component<typeof Event> {
+  <template>
+    <div class='row'>
+      <div class='column'>
+        <FieldContainer @label='Subject' @tag='label' class='field'>
+          <@fields.subject />
+        </FieldContainer>
+        <FieldContainer @label='Location' @tag='label' class='field'>
+          <@fields.location />
+        </FieldContainer>
+        <FieldContainer @label='Start' @tag='label' class='field'>
+          <div class='inner-row'>
+            <@fields.startDate />
+            <div class='divider' />
+            <@fields.startTime />
+          </div>
+        </FieldContainer>
+        <FieldContainer @label='End' @tag='label' class='field'>
+          <div class='inner-row'>
+            <@fields.endDate />
+            <div class='divider' />
+            <@fields.endTime />
+          </div>
+        </FieldContainer>
+        <FieldContainer @label='Event type' @tag='label' class='field'>
+          <@fields.description />
+        </FieldContainer>
+        <FieldContainer @label='Description' @tag='label' class='field'>
+          <@fields.description />
+        </FieldContainer>
+        <FieldContainer
+          @label='Is reminder set'
+          @tag='label'
+          class='field column'
+        >
+          <@fields.isReminderSet />
+        </FieldContainer>
+      </div>
+      <div class='column'>
+        <FieldContainer @label='Assigned to' @tag='label' class='field'>
+          <@fields.assignee />
+        </FieldContainer>
+      </div>
+    </div>
+    <div class='row'>
+      <FieldContainer @label='Location' @tag='label' class='field'>
+        <@fields.location />
+      </FieldContainer>
+      <div class='column' />
+    </div>
+    <style>
+      .row {
+        display: flex;
+        flex-direction: row;
+        justify-content: space-between;
+        padding: 0 calc(1.25rem * 1.333);
+        margin-top: 20px;
+      }
+      .inner-row {
+        width: 100%;
+        display: flex;
+        flex-direction: row;
+      }
+      .divider {
+        width: 20px;
+      }
+      .column {
+        width: 50%;
+      }
+      .column > label {
+        margin-bottom: 20px;
+      }
+    </style>
+  </template>
+}
+
+export class Event extends CardDef {
+  static displayName = 'Event form';
+  @field subject = contains(StringCard);
+  @field location = contains(StringCard);
+  @field assignee = linksTo(PersonCard);
+  @field startDate = contains(DateField);
+  @field endDate = contains(DateField);
+  @field startTime = contains(TimeCard);
+  @field endTime = contains(TimeCard);
```

Comment:
I question the need for. TimeCard, why are you not using DateTime card that already exists? Normally the time is never separated from the date. Only for very unique situations does that occur.

Reply:
If the desire is a more custom entry field where time and date are entered separately, then extend the already existing DateTime card and provide the custom template for the time entry.

Reply:
Good point. @mahdidavoodi7 can you for now use DateTime as exemplified in the Booking card

![Uploading Screenshot 2024-05-02 at 12.09.44.png…]()